src/ArraySetops.chpl

```diff
@@ -148,8 +148,12 @@ module ArraySetops
 
       var aMin = -1, aMax = -1, bMin = -1, bMax = -1;
       // we iterate over locales serially because the logic relies on the mins and maxs referring to the previous locale
-      for loc in Locales {
-        on loc {  // perform loop body computation on locale #loc
+      // use an atomic counter to ensure the previous locale finishes before the next begins (on statements are non-blocking)
+      var count: atomic int;
```

Comment:
You could also use a sync variable :)

Reply:
I did it with a sync var at first! And I think they should be equivalent. But I do like the explicitness of each locale waiting for the counter to hit their number before beginning as a guarantee that they happen both serially and in order